cmd/controller/main.go

```diff
@@ -17,178 +17,25 @@ limitations under the License.
-	configMapInformer := kubeInformerFactory.Core().V1().ConfigMaps()
-	imageInformer := cachingInformerFactory.Caching().V1alpha1().Images()
-
-	// Build all of our controllers, with the clients constructed above.
-	// Add new controllers to this array.
-	controllers := [...]*controller.Impl{
-		configuration.NewController(
-			opt,
-			configurationInformer,
-			revisionInformer,
-		),
-		revision.NewController(
-			opt,
-			revisionInformer,
-			kpaInformer,
-			imageInformer,
-			deploymentInformer,
-			coreServiceInformer,
-			endpointsInformer,
-			configMapInformer,
-		),
-		route.NewController(
-			opt,
-			routeInformer,
-			configurationInformer,
-			revisionInformer,
-			coreServiceInformer,
-			clusterIngressInformer,
-			certificateInformer,
-		),
-		labeler.NewRouteToConfigurationController(
-			opt,
-			routeInformer,
-			configurationInformer,
-			revisionInformer,
-		),
-		service.NewController(
-			opt,
-			serviceInformer,
-			configurationInformer,
-			revisionInformer,
-			routeInformer,
-		),
-		serverlessservice.NewController(
-			opt,
-			sksInformer,
-			coreServiceInformer,
-			endpointsInformer,
-		),
-	}
-	// This line asserts at compile time that the length of controllers is equal to numControllers.
-	// It is based on https://go101.org/article/tips.html#assert-at-compile-time, which notes that
-	// var _ [N-M]int
-	// asserts at compile time that N >= M, which we can use to establish equality of N and M:
-	// (N >= M) && (M >= N) => (N == M)
-	var _ [numControllers - len(controllers)][len(controllers) - numControllers]int
-
-	// Watch the logging config map and dynamically update logging levels.
-	opt.ConfigMapWatcher.Watch(logging.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, component))
-	// Watch the observability config map and dynamically update metrics exporter.
-	opt.ConfigMapWatcher.Watch(metrics.ConfigMapName(), metrics.UpdateExporterFromConfigMap(component, logger))
-	if err := opt.ConfigMapWatcher.Start(stopCh); err != nil {
-		logger.Fatalw("failed to start configuration manager", zap.Error(err))
-	}
-
-	// Start all of the informers and wait for them to sync.
-	logger.Info("Starting informers.")
-	if err := controller.StartInformers(
-		stopCh,
-		clusterIngressInformer.Informer(),
-		certificateInformer.Informer(),
-		configMapInformer.Informer(),
-		configurationInformer.Informer(),
-		coreServiceInformer.Informer(),
-		deploymentInformer.Informer(),
-		endpointsInformer.Informer(),
-		imageInformer.Informer(),
-		kpaInformer.Informer(),
-		revisionInformer.Informer(),
-		routeInformer.Informer(),
-		serviceInformer.Informer(),
-		sksInformer.Informer(),
-	); err != nil {
-		logger.Fatalw("Failed to start informers", err)
-	}
-
-	// Start all of the controllers.
-	logger.Info("Starting controllers.")
-	controller.StartAll(stopCh, controllers[:]...)
-}
-
-func flush(logger *zap.SugaredLogger) {
-	logger.Sync()
-	pkgmetrics.FlushExporter()
+	sharedmain.Main("controller",
```

Comment:
When/why will this be something else besides "controller"? in the eventing source controller pr I used "sources-controller" but advice would be nice.

Reply:
this is our main controller, which historically has used this component name (which is used e.g. to determine the log-level in config-logging)